### PR TITLE
Enable SourceLink support

### PR DESCRIPTION
**Pull request by qmfrederik**
Allows publishing of the source code of DiscUtils using a `.snupkg` package, and patching of the symbol files so that they reference the source code on GitHub.

### CODE REVIEW

**Comment by LordMike** [comment]
It seems we don't build / publish symbols? Can that be true?

File: Library/common.props [hunk 1]
```diff
@@ -6,9 +6,19 @@
     
     <PackageProjectUrl>https://github.com/DiscUtils/DiscUtils</PackageProjectUrl>
     <PackageLicenseUrl>https://github.com/DiscUtils/DiscUtils/blob/master/LICENSE.txt</PackageLicenseUrl>
-    <RepositoryUrl>https://github.com/DiscUtils/DiscUtils</RepositoryUrl>
```
Comment (LordMike) [inline Library/common.props]: RepositoryUrl is not needed?

Comment (qmfrederik) [reply Library/common.props]: Correct, SourceLink will automatically populate `RepositoryUrl` for you. One of the nice side-effects is that if someone forks this project, `RepositoryUrl` gets updated automatically and points to their fork.

**Comment by qmfrederik** [comment]
> It seems we don't build / publish symbols? Can that be true?

The file extension for symbols changed to `.snupkg`. I updated `appveyor.yml` to work with the new file format.

**Comment by qmfrederik** [comment]
Symbol files are back: https://ci.appveyor.com/project/LordMike/discutils/builds/24917403/artifacts

**Comment by LordMike** [comment]
Looks good then. Only thing I see now is that Appveyor does not seem to push symbols by default (not snupkg format, and not to Nugets symbols feed).

* [this](https://www.appveyor.com/docs/deployment/nuget/) says that Appveyor will push `symbols.nupkg` to `SymbolSource.org`
* [Nuget](https://docs.microsoft.com/en-us/nuget/create-packages/symbol-packages-snupkg#publishing-a-symbol-package) says we should push to their regular feed (`https://api.nuget.org/v3/index.json`) .. they'll sort out symbols etc from there. 

So I think appveyor config should be this .. would you agree?.. Alternatively, we replace the deployment section with some `dotnet nuget push ..` commands.

```yaml
deploy:
- provider: NuGet
  server: https://api.nuget.org/v3/index.json
  symbol_server: https://api.nuget.org/v3/index.json
  artifact: /.*\.s?nupkg/
  api_key:
    secure: 1zdVvAQJbpFDsKJS/TXQLc6GI6SlZueTSCcL7xbK6QMisvL8V5nhLFgrceLfp1Ak
  on:
    branch: nuget
```

**Comment by qmfrederik** [comment]
Yes, that sounds about right, let's find out if it works :)

**Comment by LordMike** [comment]
Also, merge in `master` .. to resolve conflicts. :)

**Pull request merged**